pkgs/by-name/ca/catppuccin-grub/package.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  flavor ? "mocha", # override with your chosen flavor
+}:
+let
+  version = "1.0.0";
+in
+stdenvNoCC.mkDerivation rec {
```

Comment:
Here, instead of of using `let` `in` to specify version, you could specify it inside the derivation, and then reference it using the `finalAttrs:` pattern, which is a way to access other attributes of a derivation without the `rec` keyword.

This is not required, but I'd recommend it

```diff
-let
-  version = "1.0.0";
-in
-stdenvNoCC.mkDerivation rec {
+
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "catppuccin-grub";
-  inherit version;
+  version = "1.0.0";

   src = fetchFromGitHub {
     owner = "catppuccin";
     repo = "grub";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     hash = "sha256-/bSolCta8GCZ4lP0u5NVqYQ9Y3ZooYCNdTwORNvR7M0=";
   };

@@ -34,4 +32,4 @@ stdenvNoCC.mkDerivation rec {
     maintainers = with lib.maintainers; [isabelroses mimvoid];
     platforms = lib.platforms.linux;
   };
-}
+})
```

Reply:
please read https://github.com/NixOS/nixpkgs/pull/316752#discussion_r1624871056